sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -135,7 +129,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * @since 1.4.0
    */
   def load(path: String): DataFrame = {
-    option("path", path).load()
+    load(Seq(path): _*) // force invocation of `load(...varargs...)`
```

Comment:
with this change path is no longer available in the options. this makes it hard (impossible?) for non-file based DataSources (not implementing FileFormat) to use load(...)

For example for elasticsearch we use:
```
sqlContext.read.format("org.elasticsearch.spark.sql").load(resource)
```
i do not think this can be implemented anymore now?